creator-node/src/routes/audiusUsers.js

```diff
@@ -1,97 +1,155 @@
 const { Buffer } = require('ipfs-http-client')
+const fs = require('fs')
 
 const models = require('../models')
 const authMiddleware = require('../authMiddleware')
 const nodeSyncMiddleware = require('../redis').nodeSyncMiddleware
 const { saveFileFromBuffer } = require('../fileManager')
-const { handleResponse, successResponse, errorResponseBadRequest } = require('../apiHelpers')
+const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError } = require('../apiHelpers')
 const { getFileUUIDForImageCID } = require('../utils')
+const { preMiddleware, postMiddleware } = require('../middlewares')
 
 module.exports = function (app) {
-  // create AudiusUser from provided metadata, and make metadata available to network
-  app.post('/audius_users', authMiddleware, nodeSyncMiddleware, handleResponse(async (req, res) => {
-    // TODO: do some validation on metadata given
-    const metadataJSON = req.body
+  /** Create AudiusUser from provided metadata, and make metadata available to network. */
+  app.post('/audius_users/metadata', authMiddleware, nodeSyncMiddleware, handleResponse(async (req, res) => {
+    // TODO - input validation
+    const metadataJSON = req.body.metadata
 
     const metadataBuffer = Buffer.from(JSON.stringify(metadataJSON))
     const { multihash, fileUUID } = await saveFileFromBuffer(req, metadataBuffer, 'metadata')
 
-    const audiusUserObj = {
-      cnodeUserUUID: req.userId,
-      metadataFileUUID: fileUUID,
-      metadataJSON: metadataJSON
-    }
+    return successResponse({ 'metadataMultihash': multihash, 'metadataFileUUID': fileUUID })
+  }))
 
-    try {
-      const coverArtFileUUID = await getFileUUIDForImageCID(req, metadataJSON.cover_photo_sizes)
-      const profilePicFileUUID = await getFileUUIDForImageCID(req, metadataJSON.profile_picture_sizes)
-      if (coverArtFileUUID) audiusUserObj.coverArtFileUUID = coverArtFileUUID
-      if (profilePicFileUUID) audiusUserObj.profilePicFileUUID = profilePicFileUUID
-    } catch (e) {
-      return errorResponseBadRequest(e.message)
+  /**
+   * Given audiusUser blockchainId, txBlockNumber, and metadataFileUUID, creates AudiusUser DB entry
+   * and associates image file entries with audiusUser. Ends audiusUser creation process.
+   */
+  app.post('/audius_users', authMiddleware, preMiddleware, nodeSyncMiddleware, handleResponse(async (req, res) => {
+    const blockchainId = req.body.blockchainUserId
```

Comment:
@SidSethi you could get the properties like const `{ blockchainId, txBlockNumber, metadataFileUUID } = req.body`

Reply:
ah very sexy, done